app/code/Magento/Catalog/Model/ResourceModel/Product/CategoryLink.php

```diff
@@ -252,4 +269,37 @@ private function analyseUpdatedLinks($deleteUpdate, $insertUpdate)
 
         return [$delete, $insert, $insertUpdate['updated']];
     }
+
+    /**
+     * Returns cache key based on product ID and category IDs
+     *
+     * Category IDs were filtered out and sorted to avoid fetching the same data from Database,
+     * when the list of categories is provided in different order.
+     *
+     * @param array $categoryIds
+     * @param int $productId
+     * @return string
+     */
+    private function getCategoryLinksCacheKey(array $categoryIds, int $productId): string
+    {
+        $categoryIds = array_filter(array_map('intval', $categoryIds));
+        sort($categoryIds);
+
+        return sprintf('%d|%s', $productId, implode(',', $categoryIds));
+    }
+
+    /**
+     * Removes ObjectCache for Category Links
+     *
```

Comment:
Please complete the comment signature 
```suggestion
     *
     ** @param string|null $cacheKey
```